pkg/feature/raw_resources.go

```diff
@@ -74,39 +59,18 @@ func (f *Feature) createResources(resources string) error {
 	return nil
 }
 
-func (f *Feature) patchResources(resources string) error {
-	splitter := regexp.MustCompile(YamlSeparator)
-	objectStrings := splitter.Split(resources, -1)
-	for _, str := range objectStrings {
-		if strings.TrimSpace(str) == "" {
-			continue
-		}
-		u := &unstructured.Unstructured{}
-		if err := yaml.Unmarshal([]byte(str), u); err != nil {
-			f.Log.Error(err, "error unmarshalling yaml")
-			return errors.WithStack(err)
-		}
-
-		ensureNamespaceIsSet(f, u)
-
-		patchAsJSON, err := yaml.YAMLToJSON([]byte(str))
+func patchResources(cli client.Client, patches []*unstructured.Unstructured) error {
+	for _, patch := range patches {
+		// Convert the individual resource patch to JSON
+		patchAsJSON, err := patch.MarshalJSON()
 		if err != nil {
-			return fmt.Errorf("error converting yaml to json: %w", err)
+			return errors.WithStack(err)
```

Comment:
I am not sure we want to stick with `.WithStack`... revert to wrap for now.

Reply:
3588b85